db/schema_applier.cc

```diff
@@ -962,13 +963,10 @@ future<> schema_applier::commit() {
     // Run func first on shard 0
     // to allow "seeding" of the effective_replication_map
     // with a new e_r_m instance.
-    co_await sharded_db.invoke_on(0, [this] (replica::database& db) { return commit_on_shard(db); });
-    co_await sharded_db.invoke_on_all([this] (replica::database& db) {
-        if (this_shard_id() == 0) {
-            return make_ready_future<>();
-        }
+    SCYLLA_ASSERT(this_shard_id() == 0);
+    commit_on_shard(sharded_db.local());
+    co_await sharded_db.invoke_on_others([this] (replica::database& db) {
         commit_on_shard(db);
-        return make_ready_future<>();
     });
```

Comment:
If we abort on commit we can as well do `invoke_on_all()`. This will also reduce latency of commit().

Reply:
invoke_on_others is here to preserve this `seeding" of the effective_replication_map` mechanism

Reply:
I see